lib/galaxy/datatypes/dataproviders/base.py

```diff
@@ -256,7 +256,6 @@ def __iter__(self):
         """
         if self.limit is not None and self.limit <= 0:
             return
-            yield
```

Comment:
since my memory was that python wouldn't allow yield + return to co-exist, it seems it will just throw an exception.

```
>>> def a():
...   yield 1
...   if True:
...     yield 2
...     return
... 
>>> a()
<generator object a at 0x7f760f80e870>
>>> q = a()
>>> q.next
<method-wrapper 'next' of generator object at 0x7f760f80e8c0>
>>> q.next()
1
>>> q.next()
2
>>> q.next()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
>>> 
```